src/common/payment.ts

```diff
@@ -289,6 +290,42 @@ export class TurboUnauthenticatedPaymentService
     return response;
   }
 
+  public async getFiatEstimateForBytes({
+    byteCount,
+    currency,
+  }: {
+    byteCount: number;
+    currency: Currency;
+  }): Promise<TurboFiatEstimateForBytesResponse> {
+    // get the price for 1 GiB
+    const wincPriceForOneGiB = await this.getUploadCosts({
+      bytes: [2 ** 30],
+    });
+    // get the winc price for one thousand fiat units (1,000 USD, or 100,000 JPY, etc.)
+    const wincPriceForOneThousandFiatUnits = await this.getWincForFiat({
+      amount: { type: currency, amount: 1_000_00 },
+    });
+
+    const wincPriceForGivenBytes = new BigNumber(wincPriceForOneGiB[0].winc)
+      .dividedBy(new BigNumber(2 ** 30))
+      .times(byteCount);
+
+    const wincPriceForOneFiatUnit = new BigNumber(
+      wincPriceForOneThousandFiatUnits.winc,
+    ).dividedBy(new BigNumber(1_000)); // convert back to 1 USD or 100 JPY
+
+    const fiatPriceForBytes = +wincPriceForGivenBytes
+      .dividedBy(wincPriceForOneFiatUnit)
+      .times(currency === 'jpy' ? 100 : 1) // Convert to zero decimal if currency is JPY
+      .toFixed(currency === 'jpy' ? 0 : 2); // Convert to string with 2 decimal places, when not zero decimal currency
+
+    return {
+      byteCount,
+      fiatEstimate: fiatPriceForBytes,
```

Comment:
nit: consider using `currency` and `amount`

Reply:
[6f98de3](https://github.com/ardriveapp/turbo-sdk/pull/262/commits/6f98de3c751a7290e04e58e8d426590dd3b3d72a)